src/Modules/ContentBlocks/Domain/ContentBlock/Command/ChangeContentBlockHandler.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace ForkCMS\Modules\ContentBlocks\Domain\ContentBlock\Command;
+
+use ForkCMS\Core\Domain\MessageHandler\CommandHandlerInterface;
+use ForkCMS\Modules\ContentBlocks\Domain\ContentBlock\ContentBlock;
+use ForkCMS\Modules\ContentBlocks\Domain\ContentBlock\ContentBlockRepository;
+use ForkCMS\Modules\Frontend\Domain\Block\Block;
+use ForkCMS\Modules\Frontend\Domain\Block\BlockRepository;
+
+final class ChangeContentBlockHandler implements CommandHandlerInterface
+{
+    public function __construct(
+        private readonly ContentBlockRepository $contentBlockRepository,
+        private readonly BlockRepository $blockRepository
+    ) {
+    }
+
+    public function __invoke(ChangeContentBlock $changeContentBlock)
+    {
+        $contentBlock = ContentBlock::fromDataTransferObject($changeContentBlock);
+
+        $this->updateFrontendBlock($contentBlock);
+
+        $previousContentBlock = $changeContentBlock->getEntity();
+        $previousContentBlock->archive();
+
+        $this->contentBlockRepository->save($previousContentBlock);
```

Comment:
no need to call the save method here, will be saved with the flush of persisting the new content block

Reply:
Don't forget to remove this line

Reply:
https://github.com/forkcms/forkcms/pull/3555/commits/78cdd19e97b2bc62ad16d0687b81072d8b8f87f2